crates/rattler_conda_types/src/version/mod.rs

```diff
@@ -567,6 +574,52 @@ impl Version {
             Cow::Borrowed(self)
         }
     }
+
+    /// Extend the version to the specified length by adding default components (0s).
+    pub fn extend_to_length(&self, length: usize) -> Result<Version, VersionExtendError> {
+        if self.segment_count() >= length {
+            return Ok(self.clone());
+        }
+
+        // copy everything up to local version
+        let mut segments = self.segments[..self.segment_count()].to_vec();
+        let components_end = segments.iter().map(|s| s.len() as usize).sum::<usize>()
+            + usize::from(self.has_epoch());
+        let mut components = self.components.clone()[..components_end].to_vec();
+
+        // unwrap is OK here because these should be fine to construct
+        let segment = Segment::new(1).unwrap().with_separator(Some('.')).unwrap();
+
+        if length > self.segment_count() {
```

Comment:
This `if` statement is not really needed right?

Reply:
removed.